doc/api/packages.md

```diff
@@ -83,6 +84,23 @@ future-proof the package in case the default type of Node.js ever changes, and
 it will also make things easier for build tools and loaders to determine how the
 files in the package should be interpreted.
 
+Node.js will refuse to load the following when passed to `node` as the
+initial input and the nearest parent `package.json` file contains a top-level
+[`"type"`][] field with a value of `"module"`:
+
+* Files whose name doesn't end in `.js`, `.mjs`, `.cjs`, or `.wasm`.
+
+Passing to `node` as the initial input when the nearest parent `package.json`
+file contains a top-level [`"type"`][] field with a value of `"commonjs"`, or
+when referenced by `require()` calls:
+
+* Files whose name ends in `.node` are interpreted as
+  compiled addon modules loaded with `process.dlopen()`.
+
+* Files whose name ends in `.json` are parsed as JSON text files.
+
+* Any other files will be treated as a [CommonJS][] module.
+
```

Comment:
Just wondering if we should move all this content to the CLI documentation perhaps?